site/en/blog/has-with-cq-m105/index.md

```diff
@@ -0,0 +1,126 @@
+---
+title: "@container + :has(): two powerful new responsive APIs landing in Chromium 105"
+description: >
+  "Container queries and :has() are a match made in responsive heaven. Luckily, both of these features are landing together in Chromium 105. This is a huge release with two highly-requested features for responsive interfaces!"
+layout: 'layouts/blog-post.njk'
+date: 2022-07-25
+hero: 'image/HodOHWjMnbNw56hvNASHWSgZyAf2/KBYS0gF3R7mdhjVBwvbn.jpg'
+alt: >
+  Image of a rocketship from SpaceX via Unsplash
+tags:
+  - chrome-105
+  - css
+authors:
+  - unakravets
+---
+
+Container queries and :has() are a match made in responsive heaven. Luckily, both of these features are landing together in Chromium 105. This is a huge release with two highly-requested features for responsive interfaces!
+
+## Container Queries: a quick summary
+
+Container queries enable developers to query a parent selector for its size and styling information, making it possible for a child element to own its responsive styling logic, no matter where it lives on a web page.
+
+-- IMG TODO --
+
+Instead of relying on the viewport for styling input such as available space, developers now have the ability to query the size of in-page elements too.. This capability means that a component owns its responsive styling logic. This makes the component much more resilient, as the styling logic is attached to it, no matter where it appears on the page.
+
+{% Aside 'caution' %}
+Container queries are [currently supported](https://caniuse.com/css-container-queries) in Chromium and Safari TP.
+{% endAside %}
+
+## Using container queries
+
+To build with container queries, you must first set containment on a parent element. Do this by setting a `container-type` on the parent container. You might have a card with an image and some text content that looks like this:
+
+{% Img src="image/HodOHWjMnbNw56hvNASHWSgZyAf2/xR7RmgS1TUZmZ9ntOFrV.png", alt="single two-column card", width="795", height="182" %}
+
+To create a container query, set `container-type` on the card container:
+
+```css
+.card-container {
+  container-type: inline-size;
+}
+```
+
+Setting the `container-type` to `inline-size` says we want to query for the inline-direction size of the parent. In latin languages like english, this would be the width of the card, since the text flows inline from left to right.
```

Comment:
```suggestion
Setting the `container-type` to `inline-size` queries the inline-direction size of the parent. In latin languages like english, this would be the width of the card, since the text flows inline from left to right.
```